### PR TITLE
Development

### PR DESCRIPTION
**Pull request by NeerNim**

### CODE REVIEW

File: enumerable.rb [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        unless yield(key, value)
+         return true
+        end
+      end
+    end
+    false
+  end
+
+  def my_count
+    return self.size unless block_give?
```
Comment (inhaq) [inline enumerable.rb]: it should be `block_given?`

**Review by inhaq — CHANGES_REQUESTED** [review]
Good job.
All methods are working as expected except `my_each` and `my_count`
 - `my_each` return single digit instead of array
 - update typo in `my_count` which results in error

Once updated, kindly resubmit fot review.

**Review by inhaq — CHANGES_REQUESTED** [review]
`my_each` is still not working properly. Let me add an example here as well.
`[1,2,3,4,5].my_each { |i| i }` returns `5` instead of `[1,2,3,4,5]`
As most of the methods depend on `my_each`, most probably they'll not work properly as well.
Once updated, kindly resubmit for review.

**Comment by NeerNim** [comment]
> `my_each` is still not working properly. Let me add an example here as well.
> `[1,2,3,4,5].my_each { |i| i }` returns `5` instead of `[1,2,3,4,5]`
> As most of the methods depend on `my_each`, most probably they'll not work properly as well.
> Once updated, kindly resubmit for review.

I had to return self in my_each method

**Comment by inhaq** [comment]
Hi @NeerNim it needs to return an array, instead of a number

**Comment by NeerNim** [comment]
> Hi @NeerNim it needs to return an array, instead of a number

Hi @inhaq, I correct it to return array instead of number

**Review by inhaq — APPROVED** [review]
Awesome, It's 👍🏻
Before merging, kindly remove comments and `include Enumerable` is not required as well.
Good luck with the next projects.

**Pull request merged**